Gordon360/Services/MembershipService.cs

```diff
@@ -475,5 +475,29 @@ private bool isPersonAlreadyInActivity(MEMBERSHIP membershipRequest)
 
             return true;
         }
+
+        /// <summary>	
+        /// Determines whether or not the given student is a Group Admin of some activity	
+        /// </summary>
+        /// <param name="studentID">The student id</param>	
+        /// <returns>true if student is a Group Admin, else false</returns>	
+        public Boolean IsGroupAdmin(int studentID)
+        {
+            IEnumerable<MembershipViewModel> memberships = GetMembershipsForStudent("" + studentID);
+
+            var membershipsWhereStudentIsGroupAdmin = _unitOfWork.MembershipRepository.Where(
```

Comment:
In the line above you GetMembershipsForStudent, but then here you're filtering the entire MembershipRepository for ones that match studentID and are Group Admin. Wouldn't it be better to use the memberships you already retrieved and filter those for ones where the student is a GroupAdmin?